File: CarHomeWidget/src/com/anod/car/home/prefs/ConfigurationInCarVolume.java
package com.anod.car.home.prefs;

import android.annotation.SuppressLint;
import android.app.ActionBar;
import android.os.Bundle;
import android.preference.Preference;
import android.view.Gravity;
import android.widget.CompoundButton;
import android.widget.CompoundButton.OnCheckedChangeListener;
import android.widget.Switch;

import com.anod.car.home.R;
import com.anod.car.home.prefs.preferences.PreferencesStorage;
import com.anod.car.home.utils.Utils;

public class ConfigurationInCarVolume extends ConfigurationActivity implements OnCheckedChangeListener {
	
	@Override
	protected boolean isAppWidgetIdRequired() {
		return false;
	}
	
	@Override
	protected int getXmlResource() {
		return R.xml.preference_incar_volume;
	}

	@Override
	protected void onCreateImpl(Bundle savedInstanceState) {
		
		if (Utils.IS_ICS_OR_GREATER) {
			initGlobalSwtich();
		}
	}

	@SuppressLint("NewApi")
	private void initGlobalSwtich() {
		ActionBar actionbar = getActionBar();
		Switch actionBarSwitch = new Switch(this);
		actionBarSwitch.setOnCheckedChangeListener(this);
		actionBarSwitch.setChecked(PreferencesStorage.isAdjustVolumeLevel(this));
		
		actionbar.setDisplayOptions(ActionBar.DISPLAY_SHOW_CUSTOM,
				ActionBar.DISPLAY_SHOW_CUSTOM);
		actionbar.setCustomView(actionBarSwitch, new ActionBar.LayoutParams(
				ActionBar.LayoutParams.WRAP_CONTENT,
				ActionBar.LayoutParams.WRAP_CONTENT, Gravity.CENTER_VERTICAL
						| Gravity.RIGHT));

	}

	@Override
	public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
		PreferencesStorage.setAdjustVolumeLevel(this, isChecked);
	}

}

File: app/src/main/java/com/anod/car/home/prefs/ConfigurationInCarVolume.kt
package com.anod.car.home.prefs

import android.widget.CompoundButton
import android.widget.CompoundButton.OnCheckedChangeListener

import com.anod.car.home.R
import info.anodsplace.carwidget.content.preferences.InCarStorage

class ConfigurationInCarVolume : ConfigurationPreferenceFragment(), OnCheckedChangeListener {

    override val isAppWidgetIdRequired: Boolean
        get() = false

    override val xmlResource: Int
        get() = R.xml.preference_incar_volume

    override val sharedPreferencesName: String
        get() = info.anodsplace.carwidget.content.preferences.InCarStorage.PREF_NAME

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
        val prefs = info.anodsplace.carwidget.content.preferences.InCarStorage.load(requireActivity())
        prefs.isAdjustVolumeLevel = isChecked
        prefs.apply()
    }

}
